contracts/DecentHats.sol

```diff
@@ -0,0 +1,395 @@
+import {IERC6551Registry} from "./interfaces/IERC6551Registry.sol";
+import {IHats} from "./interfaces/hats/full/IHats.sol";
+import {LockupLinear, Broker} from "./interfaces/sablier/full/types/DataTypes.sol";
+import {DecentAutonomousAdmin} from "./DecentAutonomousAdmin.sol";
+import {IHatsModuleFactory} from "./interfaces/hats/full/IHatsModuleFactory.sol";
+import {IHatsElectionEligibility} from "./interfaces/hats/full/IHatsElectionEligibility.sol";
+import {ModuleProxyFactory} from "@gnosis.pm/zodiac/contracts/factory/ModuleProxyFactory.sol";
+import {ISablierV2LockupLinear} from "./interfaces/sablier/ISablierV2LockupLinear.sol";
+
+contract DecentHats {
+    string public constant NAME = "DecentHats";
+    bytes32 public constant SALT =
+        0x5d0e6ce4fd951366cc55da93f6e79d8b81483109d79676a04bcc2bed6a4b5072;
+
+    struct SablierStreamParams {
+        ISablierV2LockupLinear sablier;
+        address sender;
+        address asset;
+        LockupLinear.Timestamps timestamps;
+        Broker broker;
+        uint128 totalAmount;
+        bool cancelable;
+        bool transferable;
+    }
+
+    struct TermedParams {
+        uint128 termEndDateTs;
+        address[] nominatedWearers;
+    }
+
+    struct Hat {
+        address wearer;
+        string details;
+        string imageURI;
+        SablierStreamParams[] sablierParams;
+        TermedParams[] termedParams;
+        uint32 maxSupply;
+        bool isMutable;
+        bool isTermed;
+    }
+
+    struct CreateTreeParams {
+        IHats hatsProtocol;
+        IERC6551Registry registry;
+        IHatsModuleFactory hatsModuleFactory;
+        ModuleProxyFactory moduleProxyFactory;
+        address decentAutonomousAdminMasterCopy;
+        address hatsAccountImplementation;
+        address keyValuePairs;
+        address hatsElectionEligibilityImplementation;
+        Hat adminHat;
+        Hat[] hats;
+        string topHatDetails;
+        string topHatImageURI;
+    }
+
+    /* /////////////////////////////////////////////////////////////////////////////
+                        EXTERNAL FUNCTIONS
+    ///////////////////////////////////////////////////////////////////////////// */
+    function createAndDeclareTree(CreateTreeParams calldata params) public {
+        (uint256 topHatId, address topHatAccount) = _createTopHatAndAccount(
+            params.hatsProtocol,
+            params.topHatDetails,
+            params.topHatImageURI,
+            params.registry,
+            params.hatsAccountImplementation
+        );
+
+        _updateKeyValuePairs(params.keyValuePairs, topHatId);
+
+        (uint256 adminHatId, ) = _createAdminHatAndAccount(
+            params.hatsProtocol,
+            params.registry,
+            params.moduleProxyFactory,
+            params.decentAutonomousAdminMasterCopy,
+            params.hatsAccountImplementation,
+            topHatAccount,
+            topHatId,
+            params.adminHat
+        );
```

Comment:
This function was created from `_createHatAndAccountAndMintAndStreams` and narrowed down to just Admin hat creation.